toast-tk-runtime/src/main/java/io/toast/tk/runtime/AbstractScenarioRunner.java

```diff
@@ -17,141 +13,133 @@
-    	final TestParser testParser = new TestParser();
-        ITestPage result = file == null ? testParser.readString(script, null) : testParser.parse(file.getPath());
-        final TestRunner runner = injector.getInstance(TestRunner.class);
-        if(this.presetRepoFromWebApp) {
-        	final String repoWiki = RestUtils.downloadRepositoryAsWiki();
-        	final TestParser parser = new TestParser();
-        	final ITestPage repoAsTestPageForConvenience = parser.readString(repoWiki, null);
-            runner.run(repoAsTestPageForConvenience);
-        } 
-        else if (this.localRepositoryTestPage != null) {
-            runner.run(this.localRepositoryTestPage);
-        }
-        beginTest();
-        result = runner.run(result);
-        createAndOpenReport(result);
-        endTest();
-        return result;
-    }
-
-    protected void createAndOpenReport(
-    	final ITestPage testPage
-    ) {
-    	final String generatePageHtml = htmlReportGenerator.generatePageHtml(testPage);
-    	final String path = getReportsFolderPath();
-        final String pageName = testPage.getName();
-        htmlReportGenerator.writeFile(generatePageHtml, pageName, path);
-        openReport(path, pageName);
-    }
+		tearDownEnvironment();
+
+		LOG.info("{}file(s) processed", scenarios.length);
+		RunUtils.printResult(testPages);
+	}
+
+	public final void runRemote(
+			final String... scenarios
+	) throws Exception {
+		this.presetRepoFromWebApp = true;
+		run(scenarios);
+	}
+
+	public final void runRemoteScript(
+			final String script
+	) throws Exception {
+		this.presetRepoFromWebApp = true;
+		runScript(script);
+	}
+
+	public void runLocalScript(
+			final String wikiScenario,
+			final String repoWiki,
+			final IReportUpdateCallBack callback
+	) throws IOException {
+		this.progressReporter.setReportCallBack(callback);
+		final TestParser parser = new TestParser();
+		this.localRepositoryTestPage = parser.readString(repoWiki, null);
+		runScript(wikiScenario);
+	}
+
+	private InputStream readTestFile(final String fileName) throws IOException, URISyntaxException {
+		return this.getClass().getClassLoader().getResourceAsStream(fileName);
+	}
+
+	private ITestPage runScript(final String script) throws IOException {
+		final TestParser testParser = new TestParser();
+		ITestPage result = testParser.readString(script, null);
+		return runTestPage(result);
+	}
+
+	private ITestPage runTestPage(ITestPage result) throws IOException {
+		final TestRunner runner = injector.getInstance(TestRunner.class);
+		if (this.presetRepoFromWebApp) {
+			final String repoWiki = RestUtils.downloadRepositoryAsWiki();
+			final TestParser parser = new TestParser();
+			final ITestPage repoAsTestPageForConvenience = parser.readString(repoWiki, null);
+			runner.run(repoAsTestPageForConvenience);
+		} else if (this.localRepositoryTestPage != null) {
+			runner.run(this.localRepositoryTestPage);
+		}
+		beginTest();
+		result = runner.run(result);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS1226)